test/framework/src/main/java/org/opensearch/test/rest/yaml/section/MatchAssertion.java

```diff
@@ -57,14 +59,56 @@
 public class MatchAssertion extends Assertion {
     public static MatchAssertion parse(XContentParser parser) throws IOException {
         XContentLocation location = parser.getTokenLocation();
-        Tuple<String, Object> stringObjectTuple = ParserUtils.parseTuple(parser);
-        return new MatchAssertion(location, stringObjectTuple.v1(), stringObjectTuple.v2());
+        String field = null;
+        Object expectedValue = null;
+        Double epsilon = null;
+
+        XContentParser.Token token = parser.currentToken();
+        if (token == XContentParser.Token.START_OBJECT) {
+            String currentFieldName = null;
+            while ((token = parser.nextToken()) != XContentParser.Token.END_OBJECT) {
+                if (token == XContentParser.Token.FIELD_NAME) {
+                    currentFieldName = parser.currentName();
+                } else if (token.isValue()
+                    || token == XContentParser.Token.START_OBJECT
+                    || token == XContentParser.Token.START_ARRAY
+                    || token == XContentParser.Token.VALUE_NULL) {
+                        if ("epsilon".equals(currentFieldName)) {
+                            if (parser.currentToken() == XContentParser.Token.VALUE_NUMBER) {
+                                epsilon = parser.doubleValue();
+                            } else {
+                                throw new IllegalArgumentException("epsilon must be a number");
+                            }
+                        } else {
+                            field = currentFieldName;
+                            expectedValue = XContentParserUtils.parseFieldsValue(parser);
+                        }
+                    }
+            }
+        } else {
+            // simple match: { field: value }
+            Tuple<String, Object> stringObjectTuple = ParserUtils.parseTuple(parser);
```

Comment:
I think you can try reuse `parseTuple` for the new logic. If `parser.map()` only have 1 item, it's the same. If the map contains 2, then one of it must be `epsilon` and the other is like before.